pkg/kubectl/resource_printer.go

```diff
@@ -504,6 +504,10 @@ func translateTimestamp(timestamp util.Time) string {
 }
 
 func printPod(pod *api.Pod, w io.Writer, withNamespace bool, wide bool, showAll bool, columnLabels []string) error {
+	return printPodBase(pod, w, withNamespace, wide, showAll, true, columnLabels)
+}
+
+func printPodBase(pod *api.Pod, w io.Writer, withNamespace bool, wide bool, showAll bool, showIfTerminating bool, columnLabels []string) error {
```

Comment:
Agreed that `kubectl get pod <pod_name>` should retrieve the pod regardless of its state.
The `-a` was introduced relatively recently to mimic the behavior of `docker ps -a`. By default, `get pods` would hide all terminated pods.